subsys/net/ip/tcp.c

```diff
@@ -910,6 +910,12 @@ void net_tcp_ack_received(struct net_context *ctx, u32_t ack)
 	u32_t seq;
 	bool valid_ack = false;
 
+	if (net_tcp_seq_greater(ack, ctx->tcp->send_seq)) {
```

Comment:
The call site for net_tcp_ack_received(), tcp_established() in net_context.c, will return NET_DROP in error cases. Shouldn't this happen as well for these cases?

Reply:
@lpereira has good point, if the packet is to be dropped we should somehow inform the caller (net_context.c:tcp_established()) about this so that it could return NET_DROP in this case.
@pfalcon can you check if doing that works?

Reply:
@jukkar, to avoid confusion, can you please check my comment at https://github.com/zephyrproject-rtos/zephyr/pull/5511#issuecomment-355532070 (which was intended as reply to @lpereira), and see which of 2 choices proposed you'd prefer.